src/main/java/com/google/impactdashboard/server/DataUpdater.java

```diff
@@ -1,39 +1,66 @@
 package com.google.impactdashboard.server;
 
+import com.google.cloud.logging.v2.LoggingClient.ListLogEntriesPagedResponse;
+import com.google.cloud.logging.v2.LoggingClient;
 import com.google.common.annotations.VisibleForTesting;
 import com.google.impactdashboard.data.IAMBindingDatabaseEntry;
+import com.google.impactdashboard.data.project.ProjectIdentification;
 import com.google.impactdashboard.data.recommendation.Recommendation;
+import com.google.impactdashboard.database_manager.data_read.DataReadManager;
+import com.google.impactdashboard.database_manager.data_read.DataReadManagerFactory;
 import com.google.impactdashboard.database_manager.data_update.DataUpdateManager;
 import com.google.impactdashboard.database_manager.data_update.DataUpdateManagerFactory;
+import com.google.impactdashboard.server.api_utilities.IamBindingRetriever;
 import com.google.impactdashboard.server.api_utilities.LogRetriever;
+import com.google.impactdashboard.server.api_utilities.ProjectListRetriever;
 import com.google.impactdashboard.server.api_utilities.RecommendationRetriever;
+import com.google.logging.v2.LogEntry;
 
 import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
 import java.util.List;
 
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
 /** Class for updating the information in the database from the API. */
 public class DataUpdater {
 
   private final LogRetriever logRetriever;
   private final RecommendationRetriever recommendationRetriever;
+  private final IamBindingRetriever iamRetriever;
   private final DataUpdateManager updateManager;
+  private final DataReadManager readManager;
+  private final ProjectListRetriever projectRetriever;
+  private final boolean manualUpdate;
 
   @VisibleForTesting
   protected DataUpdater(LogRetriever logRetriever, RecommendationRetriever recommendationRetriever,
-                      DataUpdateManager updateManager) {
+                      DataUpdateManager updateManager, DataReadManager readManager,
+                      IamBindingRetriever iamRetriever, ProjectListRetriever projectRetriever,
+                      boolean manualUpdate) {
```

Comment:
Generally, boolean parameters should be avoided -> prefer enums instead. 

E.g. Enum UpdateMode{
LAST_DAY,
ALL_DAYS
}

Reply:
good idea!